sw/airborne/modules/ctrl/ctrl_eff_sched_rot_wing_v3b.c

```diff
@@ -0,0 +1,516 @@
+float rot_wing_side_motors_g1_q_90[2] = ROT_WING_SCHED_G1_Q_90; 
+
+// Define polynomial constants for effectiveness values for aerodynamic surfaces
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_P
+#error "ROT_WING_SCHED_AERO_CONST_P should be defined"
+#endif
+
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_Q
+#error "ROT_WING_SCHED_AERO_CONST_Q should be defined"
+#endif
+
+#ifndef ROT_WING_SCHED_G1_AERO_CONST_R
+#error "ROT_WING_SCHED_AERO_CONST_R should be defined"
+#endif
+
+float rot_wing_aerodynamic_eff_const_g1_p[1] = ROT_WING_SCHED_G1_AERO_CONST_P;
+float rot_wing_aerodynamic_eff_const_g1_q[1] = ROT_WING_SCHED_G1_AERO_CONST_Q;
+float rot_wing_aerodynamic_eff_const_g1_r[1] = ROT_WING_SCHED_G1_AERO_CONST_R;
+
+// Define settings to multiply initial control eff scheduling values
+float lift_d_multiplier = 0.7;
+float g1_p_multiplier = 1.;
+float g1_q_multiplier = 1.;
+float g1_r_multiplier = 1.;
+float g1_t_multiplier = 1.;
+
+bool wing_rotation_sched_activated = true;
+bool pusher_sched_activated = true;
+
+float sched_pitch_hover_deg = 0.;
+float sched_pitch_forward_deg = 10.;
+float sched_lower_hover_speed = 7.;
+float sched_upper_hover_speed = 14.;
+
+float pitch_angle_set = 0;
+float pitch_angle_range = 3.;
+
+// Define filters
+#ifndef ROT_WING_SCHED_AIRSPEED_FILTER_CUTOFF
+#define ROT_WING_SCHED_AIRSPEED_FILTER_CUTOFF 0.1
+#endif
+
+Butterworth2LowPass airspeed_lowpass_filter;
+
+// Define scheduling constants
+const float k_elevator[3] = {0.4603,  -4.81466, -28.8464};
+const float k_rudder[3] = {-26.1434, -0.336403, -1.16702 };
+const float k_pusher[2] = {0.007777, -0.67521};
+
+float I_xx = 0.1184;
+float I_yy = 1.1552;
+float I_zz = 1.18;
+const float weight_sched = 6.94;
+
+// Some moved outer loop settings. TODO: make struct
+float pitch_priority_factor = 11.;
+float roll_priority_factor = 10.;
+float thrust_priority_factor = 7.;
+float pusher_priority_factor = 30.;
+
+bool hover_motors_active = true;
+bool bool_disable_hover_motors = false;
+
+float horizontal_accel_weight = 10.;
+float vertical_accel_weight = 10.;
+
+inline void update_inertia(float *cosr2, float *sinr2);
+inline void update_hover_motor_effectiveness(float *sk, float *cosr, float *sinr, float *airspeed_f);
+inline void update_elevator_effectiveness(int16_t *elev_pprz, float *airspeed, float *airspeed2, float *pp_scaled);
+inline void update_rudder_effectiveness(float *airspeed2, float *pp_scaled, float *T_mean_scaled, float *cosr);
+//inline void update_left_aileron_effectiveness(float *airspeed2, float *sinr);
+inline void update_aileron_effectiveness(float *airspeed2, float *sinr);
+inline void update_flap_aileron_effectiveness(float *airspeed2, float *sinr);
+inline void update_pusher_effectiveness(float *airspeed_f, float pusher_cmd_filt);
+inline void schedule_pref_pitch_angle_deg(float wing_rot_deg);
+inline void schedule_pitch_priority_factor(float wing_rot_deg);
+inline void schedule_liftd(float *airspeed2, float *sinr, float wing_rot_deg);
```

Comment:
most of this functions have pointers as parameters, but they don't modify the values, so it doesn't look necessary.

Reply:
ok